scrapy/contrib/periodicstats/collectors.py

```diff
@@ -0,0 +1,157 @@
+from collections import defaultdict
+import re
+
+from twisted.internet import task
+
+from scrapy.statscol import MemoryStatsCollector
+from scrapy.exceptions import NotConfigured
+from scrapy.utils.misc import load_object
+from scrapy.contrib.periodicstats.observers import PeriodicStatsObserver
+from scrapy.contrib.periodicstats.pipelines import PeriodicStatsPipeline
+
+
+DEFAULT_PERIODIC_STATS_ENABLED = True
+DEFAULT_PERIODIC_STATS_EXPORT_ALL = False
+DEFAULT_PERIODIC_STATS_EXPORT_ALL_INTERVAL = 60
```

Comment:
These defaults should be defined on scrapy/settings/default_settings.py (w/o the DEFAULT_ prefix). The getter functions on the Settings instance on the crawler will take those defaults if they aren't overridden somewhere.